enum ResType
{
    Mapping,
    Configure,
    Strategy
}